websites/Z/ZombsRoyale.io/presence.ts

```diff
@@ -0,0 +1,108 @@
+
+  if (currentGameState === "loading" || currentGameState === "UiLoadingOverlay") {
+    presenceData.details = "Loading...";
+  } else if (currentGameState === "MainMenu") {
+    presenceData.details = "Lurking in main menu";
+  } else if (currentGameState === "Cosmetics" || currentGameState === "UiCosmeticSelectorOverlay") {
+    presenceData.details = "Viewing cosmetics";
+  } else if (currentGameState === "Profile") {
+    presenceData.details = "Viewing their profile";
+  } else if (currentGameState === "Shop" || currentGameState === "UiPreviewPackOverlay") {
+    presenceData.details = "Browsing Shop";
+  } else if (currentGameState === "Friends" || currentGameState === "UiFriendAddOverlay") {
+    presenceData.details = "Viewing their friends";
+  } else if (currentGameState === "Leaderboards") {
+    presenceData.details = "Browsing leaderboards";
+  } else if (currentGameState === "UiLoginOverlay") {
+    presenceData.details = "Loging in";
+  } else if (currentGameState === "UiSettingsOverlay") {
+    presenceData.details = "Changing settings";
+  } else if (currentGameState === "UiSeasonPurchaseOverlay") {
+    presenceData.details = "Buying a Battle Pass";
+  } else if (currentGameState.startsWith("UiSeason") || currentGameState === "UiPreviewItemOverlay") {
+    presenceData.details = "Viewing Battle Pass";
+  } else if (currentGameState === "UiChallengesOverlay") {
+    presenceData.details = "Viewing challenges";
+  } else if (currentGameState === "UiBuyGemsOverlay") {
+    presenceData.details = "Buying gems";
+  } else if (currentGameState === "VideoAd") {
+    presenceData.details = "Watching video ad";
+  } else if (currentGameState === "Countdown" || currentGameState === "Lobby") {
+    presenceData.details = "In Lobby";
+    presenceData.state = "Waiting for game to start";
+  } else if (currentGameState === "Game" || currentGameState === "UiReportPlayerOverlay" || currentGameState === "UiLeaveOverlay") {
+    presenceData.details = "In Game";
+  } else if (currentGameState === "Plane") {
+    presenceData.details = "In Game";
+    presenceData.state = "Flying plane";
+  } else if (currentGameState === "Parachute") {
+    presenceData.details = "In Game";
+    presenceData.state = "Parachuting";
+  } else if (currentGameState === "Dead" || currentGameState === "UiGameOver" || currentGameState === "UiSpectator" || currentGameState === "UiFeedbackOverlay") {
+    presenceData.details = "In Game";
+    presenceData.state = "Dead";
+  } else if (currentGameState === "UiMapOverlay") {
+    presenceData.details = "In Game";
+    presenceData.state = "Viewing map";
+  }else {
+    presence.error("Unknown state: " + currentGameState);
+  }
+
+  const playing = presenceData.details === "In Game" || presenceData.details === "In Lobby";
+
+  if(playing && matchStart === null)
+  {
+    matchStart = Math.floor(Date.now() / 1000);
+  }
+  else if(!playing && matchStart !== null)
+  {
+    matchStart = null;
+  }
```

Comment:
[Start a Codeblock's Curly Braces on the Same Line](https://github.com/goldbergyoni/nodebestpractices#-33-start-a-codeblocks-curly-braces-on-the-same-line)